### PR TITLE
Added a GoAccess template

### LINKED ISSUE #306: CA Template Request - GoAccess
Goaccess that we have in CA template is old version from https://hub.docker.com/r/gregyankovoy/goaccess/dockerfile
updated year ago

Official is:
https://hub.docker.com/r/allinurl/goaccess
(https://github.com/allinurl/goaccess)

#### Post any additional information about the project that you know is important 
About
GoAccess is a real-time web log analyzer and interactive viewer that runs in a terminal in *nix systems or through your browser.

### PR DESCRIPTION
**Pull request by bravohotel91**
Created a GoAccess template in response to Issue #306 

Closes #306

### CODE REVIEW

**Comment by nwithan8** [comment]
There is an existing template, but it looks like it's someone who made their own container that simply builds the original application from source: https://unraid.net/community/apps?q=goaccess#r

It's several years out of date, so I'm fine with introducing a more up-to-date template that will actually use the original source code.

File: templates/goaccess.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  <Registry>https://hub.docker.com/r/allinurl/goaccess/</Registry>
+  <Network>bridge</Network>
+  <Shell>sh</Shell>
+  <Privileged>false</Privileged>
+  <Support>https://github.com/allinurl/goaccess/issues/</Support>
+  <Project>https://goaccess.io/</Project>
+  <Overview>GoAccess is a real-time web log analyzer and interactive viewer that runs in a terminal in *nix systems or through your browser.</Overview>
+  <WebUI>http://[IP]:[PORT:7890]</WebUI>
+  <PostArgs>/var/log/access.log --log-format=COMBINED -o /var/www/goaccess/index.html --real-time-html --port=7890</PostArgs>
+  <Config Name="Log Location" Target="/var/log" Default="" Mode="ro" Description="Access Log Location" Type="Path" Display="always" Required="true" Mask="false"></Config>
```
Comment (nwithan8) [inline templates/goaccess.xml:14]:
```suggestion
  <Config Name="Log Location" Target="/var/log" Default="" Mode="rw" Description="Access Log Location" Type="Path" Display="always" Required="true" Mask="false"></Config>
```

Container will need to be able to write to the log directory

Comment (nwithan8) [inline templates/goaccess.xml:14]:
```suggestion
  <Config Name="Log Location" Target="/var/log" Default="" Mode="ro" Description="Location for an access.log file" Type="Path" Display="always" Required="true" Mask="false"></Config>
```

File: templates/goaccess.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <Config Name="Report Location" Target="/var/www/goaccess" Default="" Mode="rw" Description="Report Location" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/goaccess/reporting</Config>
```
Comment (nwithan8) [inline templates/goaccess.xml:15]:
```suggestion
  <Config Name="Report Location" Target="/var/www/goaccess" Default="" Mode="rw" Description="Report Output Location" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/goaccess/reports</Config>
```

**Review by nwithan8 — CHANGES_REQUESTED** [review]

**Comment by bravohotel91** [comment]
@nwithan8 - I appreciate the check. I was initially going to comment and state that there were already some out there, but they are all not using the official container of the project and out-of-date.

File: templates/goaccess.xml (hunk 1 above)
Comment (bravohotel91) [reply templates/goaccess.xml]: wouldnt let me commit this one. I committed separately.

**Review by nwithan8 — APPROVED** [review]

**Review by nwithan8 — CHANGES_REQUESTED** [review]
Please add an icon and point to it in the template.

**Review by nwithan8 — APPROVED** [review]

File: templates/goaccess.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  <Shell>sh</Shell>
+  <Privileged>false</Privileged>
+  <Support>https://github.com/allinurl/goaccess/issues/</Support>
+  <Project>https://goaccess.io/</Project>
+  <Overview>GoAccess is a real-time web log analyzer and interactive viewer that runs in a terminal in *nix systems or through your browser.</Overview>
+  <WebUI>http://[IP]:[PORT:7890]</WebUI>
+  <PostArgs>/var/log/access.log --log-format=COMBINED -o /var/www/goaccess/index.html --real-time-html --port=7890</PostArgs>
+  <Config Name="Log Location" Target="/var/log" Default="" Mode="rw" Description="Location for an access.log file" Type="Path" Display="always" Required="true" Mask="false"></Config>
+  <Config Name="Report Location" Target="/var/www/goaccess" Default="" Mode="rw" Description="Report Output Location" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/goaccess/reports</Config>
+  <Config Name="Appdata Location" Target="/config" Default="/mnt/user/appdata/goaccess" Mode="rw" Description="App Data Location" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/goaccess</Config>
```
Comment (Roxedus) [inline templates/goaccess.xml:17]: /config does not look like a used path for this application?

Comment (bravohotel91) [reply templates/goaccess.xml:17]: That is correct. my mistake. i thought i saw one in the docs for it but i guess not

File: templates/goaccess.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  <Icon>https://raw.githubusercontent.com/selfhosters/unRAID-CA-templates/master/templates/img/goaccess.png</Icon>
+  <Registry>https://hub.docker.com/r/allinurl/goaccess/</Registry>
+  <Network>bridge</Network>
+  <Shell>sh</Shell>
+  <Privileged>false</Privileged>
+  <Support>https://github.com/allinurl/goaccess/issues/</Support>
+  <Project>https://goaccess.io/</Project>
+  <Overview>GoAccess is a real-time web log analyzer and interactive viewer that runs in a terminal in *nix systems or through your browser.</Overview>
+  <WebUI>http://[IP]:[PORT:7890]</WebUI>
+  <PostArgs>/var/log/access.log --log-format=COMBINED -o /var/www/goaccess/index.html --real-time-html --port=7890</PostArgs>
```
Comment (Roxedus) [inline templates/goaccess.xml:14]: no port mapping to expose the port

**Review by Roxedus — CHANGES_REQUESTED** [review]

**Comment by bravohotel91** [comment]
@nwithan8  changes made to add port mapping

**Review by Roxedus — APPROVED** [review]

**Comment by nwithan8** [comment]
Had to rebase to fix the CI linter

**Pull request merged**